source/opt/ir_context.h

```diff
@@ -1208,6 +1212,25 @@ Instruction* IRContext::GetMemberName(uint32_t struct_type_id, uint32_t index) {
   return nullptr;
 }
 
+void IRContext::CloneNames(uint32_t old_id, uint32_t new_id, uint32_t new_len) {
+  std::vector<std::unique_ptr<Instruction>> names_to_add;
+  auto names = GetNames(old_id);
+  for (auto n : names) {
+    Instruction* old_name_inst = n.second;
+    if (old_name_inst->opcode() == spv::Op::OpMemberName && new_len != 0) {
+      auto midx = old_name_inst->GetSingleWordInOperand(1);
+      if (midx >= new_len) continue;
```

Comment:
Are we sure the OpMemberName is always after a potential OpName? From the spec it seems both should be after OpString etc, but don't seem to have a struct requirement for this order. So if OpName is written after, then this will fail.
So maybe don't early break, but iterate on all, just checking for indices?

Reply:
I don't think this should fail. I am not breaking, I am continuing ie just skipping OpMemberName with out of bounds indices. Right?

Reply:
Oh sorry, I thought it was a break, not a continue.. my bad!